pkg/server/api/svid/v1/service.go

```diff
@@ -57,6 +61,25 @@ type Service struct {
 	ca ca.ServerCA
 	ef AuthorizedEntryFetcher
 	td spiffeid.TrustDomain
+	ds datastore.DataStore
+}
+
+func (s *Service) parseAndCheckCSR(ctx context.Context, csrBytes []byte) (*x509.CertificateRequest, error) {
```

Comment:
How would you feel about making this a standalone function? It doesn't use any state on the `*Service`. It should also be grouped below the exported methods.